samza-core/src/main/java/org/apache/samza/config/ApplicationConfig.java

```diff
@@ -46,6 +46,11 @@ public enum ApplicationMode {
     BATCH
   }
 
+  public enum ApplicationApiType {
+    LEGACY,
+    LOW_LEVEL,
+    HIGH_LEVEL
+  }
```

Comment:
Please introduce this `enum` in samza-api module instead?

Reply:
The reason to do so even if its internally used is to make sure any changes to this enum should be treated as API changes and would need to be treated as backward compatible as this will be used in the coordinator stream configuration which are persisted and read across different versions of samza.